api/src/main/java/jakarta/transaction/TransactionController.java

```diff
@@ -0,0 +1,186 @@
+     * has been marked so that the only possible outcome of the transaction
+     * is to roll back the transaction.
+     *
+     * @return {@code true} if the current transaction has been marked for
+     *    rollback, or {@code false} otherwise.
+     */
+    boolean isRollbackOnly();
+
+    /**
+     * Obtain the status of the transaction associated with the current thread.
+     *
+     * @return The transaction status. If no transaction is associated with
+     *    the current thread, this method returns the Status.NoTransaction
+     *    value.
+     *
+     * @exception IllegalStateException Thrown if the transaction manager
+     *    encounters an unexpected error condition.
+     *
+     */
+    int getStatus();
+
+    /**
+     * Modify the timeout value that is associated with transactions started
+     * by the current thread with the {@link #begin} method.
+     *
+     * <p> If an application has not called this method, the transaction
+     * service uses some default value for the transaction timeout.
+     *
+     * @param seconds The value of the timeout in seconds. If the value is zero,
+     *        the transaction service restores the default value. If the value
+     *        is negative a SystemException is thrown.
+     *
+     * @exception IllegalStateException Thrown if the transaction manager
+     *    encounters an unexpected error condition.
+     *
+     */
+    void setTimeout(int seconds);
+
+    /**
+     * The timeout value that is associated with transactions started by
+     * the current thread with the {@link #begin} method.
+     *
+     * <p> If an application has not called this method, the transaction
+     * service uses some default value for the transaction timeout.
+     *
+     * @return The timeout in seconds
+     *
+     * @exception IllegalStateException Thrown if the transaction manager
+     *    encounters an unexpected error condition.
+     *
+     */
+    int getTimeout();
+
+    /**
+     * Run the given {@linkplain Runnable operation} in a transaction.
+     * If there is no active transaction associated with the current
+     * thread, {@linkplain #begin} a new transaction before running
+     * the operation. If the operation completes without throwing an
+     * exception, {@linkplain #commit} the transaction. If the
+     * operation does throw an exception, {@linkplain #rollback} the
+     * transaction and rethrow the exception. Otherwise, if there is
+     * already an active transaction associated with the current
+     * thread, simply call the operation.
+     *
+     * @param work Some work to be performed in a transaction
+     *
+     * @exception IllegalStateException Thrown if the current
+     * {@linkplain #getStatus status} is anything other than
+     * {@link Status#STATUS_ACTIVE STATUS_ACTIVE} or
+     * {@link Status#STATUS_MARKED_ROLLBACK STATUS_MARKED_ROLLBACK}
+     * or if committing the transaction fails.
```

Comment:
We might want to be a bit more prescriptive here and say:

```suggestion
     * @exception IllegalStateException Thrown if the current
     * {@linkplain #getStatus status} is anything other than
     * {@link Status#STATUS_ACTIVE STATUS_ACTIVE} or
     * {@link Status#STATUS_MARKED_ROLLBACK STATUS_MARKED_ROLLBACK}
     * or if committing the transaction fails.
     * In the case of a rollback or heuristic decision, the thrown
     * {@link IllegalStateException} must contain a wrapped
     * {@linkplain Throwable#getCause cause} of type
     * {@link RollbackException}, {@link HeuristicMixedException}, or
     * {@link HeuristicRollbackException}, as appropriate.
```